### DISCUSSION TITLE
make process is stuck

### DISCUSSION THREAD

**Discussion by thiesschneider** [Freetz-NG/freetz-ng#1049] [Allgemein]
Hi,
I am using the docker image from pfichtner-freetz. I want to make an image for the Fritz!Box 7490
I started the docker image, made an update, cloned the repository and ran the config.
Then I ran `make` and after some time the process was stuck at the following log output:

```
CC="gcc" \
	CXX="g++" \
	CFLAGS=" -D_GNU_SOURCE -fno-stack-protector --std=c++03" \
	CXXFLAGS=" -D_GNU_SOURCE -fno-stack-protector --std=c++03" \
	cmake . \
	-DCMAKE_C_COMPILER_TARGET=x86_64-pc-linux-gnu \
	-DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-linux-gnu \
	-DCMAKE_INSTALL_PREFIX=/usr \
	-DENABLE_DOCUMENTATION=OFF \
	-DENABLE_TESTING=OFF \
	-DREDIS_STORAGE_BACKEND=OFF \
	-DCMAKE_MESSAGE_LOG_LEVEL=status \
	 \
);
Re-run cmake no build system arguments
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/gcc
-- The ASM_MASM compiler identification is unknown
-- Found assembler: ml
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
```

This is the state after ~4 hours and the machine it is running on is a mac mini m1 with docker running on colima.
Is this expected, if not, how can I fix it?

**Comment by fischer-andreas** [comment]
The Apple M1 chip is based on ARM architecture, whereas pfichtner-freetz, is designed for x86_64 (AMD64) architecture. The issue is with Colima/Colima's configuration or the x86_64 emulation.

**Comment by thiesschneider** [comment]
Ok, I ran it with rosetta as well as with the argument`--platform linux/amd64` as suggested on the pfichtner-freetz [readme](https://github.com/pfichtner/pfichtner-freetz#some-hints-if-youre-new-to-docker).

> - When running on on a mac with arm cpu (M1/M2) pass --platform linux/amd64 as additional argument to use x86 image

As far as I understand this should not happen then. Is it possible that there is another solution for it?
